src/classes/user.py

```diff
@@ -1,8 +1,86 @@
-# pylint: disable=too-few-public-methods
+class InvalidUserData(Exception):
+    pass
+
 class User() :
-    """Class for user object"""
-    def __init__(self,idx: int,username: str,score :int,rank: int) -> None:
-        self.idx = idx
-        self.username = username
-        self.score = score
-        self.rank = rank
+    """Class for the User object"""
+    def __init__(self, data: dict):
+        """
+            idx: int,
+            username: str,
+            score :int,
+            rank: int,
+            solves: int,
+        """
+        parsed_data = User.parse_rootme_user_data(data)
+
+        # see keys and parse_rootme_user_data for attribute list and values
+        for key, value in parsed_data.items():
+            setattr(self, key, value)
+
+        self.nb_new_solves = 0
+
+    @staticmethod
+    def keys():
+        return ["idx", "username", "score", "rank", "nb_solves"]
+
+    # TODO: move these static methods to rootme_api, it make more sense
+    # or create a Parser object? we could then have a RootMeParser and a HTBParser 
+    # but then what's the point of the API Manager ?
+    @staticmethod
+    def parse_rootme_user_data(data):
+        """A static method that returns a tuple of few user data extracted from raw RootMe API data"""
+
+        idx = int(data["id_auteur"])
+        username = data["nom"]
+        score = int(data["score"])
+        rank = int(data["position"])
+        nb_solves = len(data["validations"])
+
+        keys = User.keys()
+        return {keys[0]: idx, keys[1]: username, keys[2]: score, keys[3]: rank, keys[4]: nb_solves}
+
+    @staticmethod
+    def parse_rootme_user_solves_and_yield(data):
+        validations = data["validations"]
+
+        # This iterator yields ALL solves, not only new ones
+        global_solves_iterator = iter(validations)
+
+        while True:
```

Comment:
Why don't you use a for loop?

Reply:
because I'm dumb :smiling_face_with_tear: I'll change it!